src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -188,46 +198,80 @@ private void EnqueueProcessSnapshot(Document newDocument)
                 }
             }
 
-            private async Task EnqueueProcessSnapshotWorkerAsync(Document document, CancellationToken cancellationToken)
+            private async Task EnqueueProcessSnapshotWorkerAsync(Document currentDocument, CancellationToken cancellationToken)
             {
                 // we will enqueue new one soon, cancel pending refresh right away
                 _reportChangeCancellationSource.Cancel();
 
-                var newText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
-                var snapshot = newText.FindCorrespondingEditorTextSnapshot();
-                if (snapshot == null)
+                var currentText = await currentDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                var currentSnapshot = currentText.FindCorrespondingEditorTextSnapshot();
+                if (currentSnapshot == null)
                 {
                     // It's possible that we're seeing a notification for an update that happened
                     // just before the file was opened, and so the document we're given is still the
                     // old one.
                     return;
                 }
 
-                var latencyTracker = new RequestLatencyTracker(SyntacticLspLogger.RequestType.SyntacticTagger);
-                using (latencyTracker)
+                var service = TryGetClassificationService(currentSnapshot);
+                object currentCachedData = null;
+                var changedSpan = currentSnapshot.GetFullSpan();
+                if (service != null)
                 {
-                    // preemptively parse file in background so that when we are called from tagger from UI thread, we have tree ready.
-                    // F#/typescript and other languages that doesn't support syntax tree will return null here.
-                    _ = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    var latencyTracker = new RequestLatencyTracker(SyntacticLspLogger.RequestType.SyntacticTagger);
+                    using (latencyTracker)
+                    {
+                        // preemptively allow the classification service to compute and cache data for this file.  For
+                        // example, in C# and VB we will parse the file so that are called from tagger from UI thread,
+                        // we have the root of the tree ready to go.
+                        currentCachedData = await service.GetDataToCacheAsync(currentDocument, cancellationToken).ConfigureAwait(false);
+
+                        // Query the service to determine waht span of the document actually changed and should be
+                        // reclassified in the host editor.
+                        changedSpan = await GetChangedSpanAsync(currentDocument, currentSnapshot, service, cancellationToken).ConfigureAwait(false);
+                    }
```

Comment:
```suggestion
                    using var latencyTracker = new RequestLatencyTracker(SyntacticLspLogger.RequestType.SyntacticTagger);
                    
                    // preemptively allow the classification service to compute and cache data for this file.  For
                    // example, in C# and VB we will parse the file so that are called from tagger from UI thread,
                    // we have the root of the tree ready to go.
                    currentCachedData = await service.GetDataToCacheAsync(currentDocument, cancellationToken).ConfigureAwait(false);

                    // Query the service to determine waht span of the document actually changed and should be
                    // reclassified in the host editor.
                    changedSpan = await GetChangedSpanAsync(currentDocument, currentSnapshot, service, cancellationToken).ConfigureAwait(false);
```

Reply:
Much nicer.  Thanks!